python/cuml/cuml/cluster/kmeans.pyx

```diff
@@ -215,9 +215,82 @@ class KMeans(UniversalBase,
         },
     }
 
-    _cpu_estimator_import_path = 'sklearn.cluster.KMeans'
-    labels_ = CumlArrayDescriptor(order='C')
-    cluster_centers_ = CumlArrayDescriptor(order='C')
+    @classmethod
+    def _get_param_names(cls):
+        return [
+            *super()._get_param_names(),
+            "n_init",
+            "oversampling_factor",
+            "max_samples_per_batch",
+            "init",
+            "max_iter",
+            "n_clusters",
+            "random_state",
+            "tol",
+            "convert_dtype"
+        ]
+
+    @classmethod
+    def _params_from_cpu(cls, model):
+        init = model.init
+        if callable(init):
+            raise UnsupportedOnGPU
+        elif isinstance(init, str) and init == "k-means++":
+            init = "scalable-k-means++"
```

Comment:
I think we should explicitly list all of the init methods that we support and otherwise fallback.